tools/packaging/static-build/virtiofsd/build-static-virtiofsd.sh

```diff
@@ -40,4 +40,43 @@ pull_virtiofsd_released_binary() {
     popd
 }
 
-pull_virtiofsd_released_binary
+init_env() {
+   case ${ARCH} in
+     "aarch64")
+       LIBC="musl"
+     ;;
+     "ppc64le")
+       LIBC="gnu"
+     ;;
+     "s390")
+       LIBC="gnu"
+  esac
+
+    ARCH_LIBC=${ARCH}-linux-${LIBC}
+}
+	   
+build_virtiofsd_from_source() {
+   echo "build viriofsd from source"
+   init_env
+
+   virtiofsd_url=$(get_from_kata_deps "externals.virtiofsd.url")
+
+   git clone ${virtiofsd_url} virtiofsd
+   pushd virtiofsd
+
+   RUSTFLAGS='-C target-feature=+crt-static -C link-self-contained=yes'
+   LIBSECCOMP_LINK_TYPE=static
+   LIBSECCOMP_LIB_PATH=/usr/lib/${ARCH_LIBC}
+   LIBCAPNG_LINK_TYPE=static
+   LIBCAPNG_LIB_PATH=/usr/lib/${ARCH_LIBC}
+   
+   cargo build --release --target $(uname -m)-unknown-linux-${LIBC}
```

Comment:
Hi @jongwu, firstly, Thank you for the PR! 

Here, using `$uname -m` won't work for `ppc64le` since the rust arch is `powerpc64le` causing the build to fail since the target wouldn't exist.
Can we use something from here - https://github.com/kata-containers/tests/blob/main/.ci/kata-arch.sh#L48?

PS: Tested the build on `ppc64le`, it works fine.

Reply:
> Hi @jongwu, firstly, Thank you for the PR!
> 
> Here, using `$uname -m` won't work for `ppc64le` since the rust arch is `powerpc64le`, thereby causing the build to fail as the target wouldn't exist. e: We already have `ARCH` declared above, it could be used instead of `uname -m` and probably a condition can be added for `ppc64le` for changing the arch?
> 

What about getting the ARCH from `rustc` instead of `uname -m` ? Something like the following (or nicer :smiley:) :

```
ARCH=$(rustc -Vv | grep ^host: | cut -d' ' -f2 | cut -d- -f1)
```



> PS: Tested the build on `ppc64le`, it works fine.

Reply:
We already have an `arch_to_golang` function in https://github.com/kata-containers/kata-containers/blob/c88a48be216db18d40ebc6c396c75da10d3857f3/tools/packaging/scripts/lib.sh#L82-L93

We could expand lib.sh and add something like `arch_to_rust`.

Reply:
>we can just modify "ARCH" in "init_env" in this PR for ppc64le. Am I right? @Amulyam24

right, thanks @jongwu!